app/src/main/java/me/hufman/androidautoidrive/music/MusicMetadata.kt

```diff
@@ -82,6 +82,12 @@ data class MusicMetadata(val mediaId: String? = null,
 					extras = desc.extras
 			)
 		}
+
+		fun copy(other: MusicMetadata): MusicMetadata {
```

Comment:
One pattern which might be fun is to allow the `copy` function to accept some optional args. For example, when the CombinedMusicAppController is updating the queueId (which the data class pattern enabled: `copy(queueId = queueController.getMetadata()?.queueId)`)
This copy function could have an optional `queueId: Long? = null` (and all the rest, for completeness), and then the next line would have each parameter specified as `..., queueId = queueId ?: other.queueId, ...`

This removes the potential to forget to update the other copy implementations hiding elsewhere in the code, by only having this single place to update if MusicMetadata gains new parameters in the future.

Reply:
I like that idea, I was going to extend this copy method to be more like the original data class copy but then got sidetracked and forgot to come back around to it. Updated to have those optional arguments

Reply:
It looks really elegant now :) I hate to be picky, can you convert it to using smart tabs, with physical tab characters up to the function body and then space characters after the tabs for the line continuation? See my previous comment about how to set up the editor to reveal the secret whitespace

Reply:
It does! No worries about being picky, I'd rather have my code formatted correctly going into master. I updated the line continuation to utilize smart tabs so now I think it should be correct!